test_bitcoin: ./chain.h:261: uint256 CBlockIndex::GetBlockHash() const: Assertion `phashBlock != nullptr' failed.

https://cirrus-ci.com/task/6024293113397248?logs=ci#L3199

```
Simulating node restart
2023-03-23T20:11:01.824273Z (mocktime: 2020-08-31T15:36:02Z) [test] [logging/timer.h:58] [Log] [bench] FlushStateToDisk: write block and undo data to disk started
2023-03-23T20:11:01.824318Z (mocktime: 2020-08-31T15:36:02Z) [test] [logging/timer.h:58] [Log] [bench] FlushStateToDisk: write block and undo data to disk completed (0.03ms)
2023-03-23T20:11:01.824330Z (mocktime: 2020-08-31T15:36:02Z) [test] [logging/timer.h:58] [Log] [bench] FlushStateToDisk: write block index to disk started
2023-03-23T20:11:01.825668Z (mocktime: 2020-08-31T15:36:02Z) [test] [logging/timer.h:58] [Log] [bench] FlushStateToDisk: write block index to disk completed (1.33ms)
2023-03-23T20:11:01.825686Z (mocktime: 2020-08-31T15:36:02Z) [test] [logging/timer.h:58] [Log] [bench] FlushStateToDisk: write coins cache to disk (0 coins, 0kB) started
2023-03-23T20:11:01.825704Z (mocktime: 2020-08-31T15:36:02Z) [test] [txdb.cpp:167] [BatchWrite] [coindb] Writing final batch of 0.00 MiB
2023-03-23T20:11:01.825719Z (mocktime: 2020-08-31T15:36:02Z) [test] [txdb.cpp:169] [BatchWrite] [coindb] Committed 0 changed transaction outputs (out of 0) to coin database...
2023-03-23T20:11:01.825732Z (mocktime: 2020-08-31T15:36:02Z) [test] [logging/timer.h:58] [Log] [bench] FlushStateToDisk: write coins cache to disk (0 coins, 0kB) completed (0.03ms)
2023-03-23T20:11:01.825744Z (mocktime: 2020-08-31T15:36:02Z) [test] [validationinterface.cpp:249] [ChainStateFlushed] [validation] Enqueuing ChainStateFlushed: block hash=696e92821f65549c7ee134edceeeeaaa4105647a3c4fd9f298c0aec0ab50425c
Performing Load/Verify/Activate of chainstate
test_bitcoin: ./chain.h:261: uint256 CBlockIndex::GetBlockHash() const: Assertion `phashBlock != nullptr' failed.
make[3]: *** [Makefile:21821: test/validation_chainstatemanager_tests.cpp.test] Error 1

--- a/src/test/validation_chainstatemanager_tests.cpp
+++ b/src/test/validation_chainstatemanager_tests.cpp
@@ -367,10 +367,13 @@ struct SnapshotTestSetup : TestChain100Setup {
 
         BOOST_TEST_MESSAGE("Simulating node restart");
         {
-            LOCK(::cs_main);
             for (Chainstate* cs : chainman.GetAll()) {
+                LOCK(::cs_main);
                 cs->ForceFlushStateToDisk();
             }
+            // Process all callbacks referring to the old manager before wiping it.
+            SyncWithValidationInterfaceQueue();
+            LOCK(::cs_main);
             chainman.ResetChainstates();
             BOOST_CHECK_EQUAL(chainman.GetAll().size(), 0);
             const ChainstateManager::Options chainman_opts{

